tests/network/conftest.py

```diff
@@ -218,7 +218,11 @@ def network_sanity(
     failure_msgs = []
     collected_tests = request.session.items
 
-    def _verify_multi_nic():
+    def _verify_multi_nic(request=request):
+        marker_args = request.config.getoption("-m")
+        if marker_args and "single_nic" in marker_args and "not single_nic" not in marker_args:
```

Comment:
I do not like this condition, it blows my mind, however, I do not have a nicer solution for now.
I hope we can find something nicer in a follow up.

Reply:
Adding the answer I gave in our discussion in slack:
If the `-m` was not sent, then `getoption` doesn't return an empty string, but rather `False`.
See the example below:
```
ipdb> request.config.getoption("-m")
'single_nic'
ipdb> request.config.getoption("-l")
False
```
So if I start by checking "single_nic" in `marker_args` when `marker_args` is False - the whole run will fail.